tests/test_apply.py

```diff
@@ -91,51 +91,11 @@ def mimic_execution_for_apply(device):
 class TestApplyPureState:
     """Test application of PennyLane operations on the pure state simulator."""
 
-    def test_basis_state(self, shots, tol):
+    @pytest.mark.parametrize("state", [[1, 0, 0, 0], [0, 0, 1, 0]])
+    def test_basis_state(self, shots, tol, state):
         """Test basis state initialization"""
         dev = SimulatorDevice(4, shots=shots)
-        state = np.array([0, 0, 1, 0])
-
-        with mimic_execution_for_apply(dev):
-            dev.apply([qml.BasisState(state, wires=[0, 1, 2, 3])])
-
-        res = dev._state
-
-        expected = np.zeros([2**4])
-        expected[np.ravel_multi_index(state, [2] * 4)] = 1
-        assert np.allclose(res, expected, **tol)
-
-    @pytest.mark.parametrize(
-        "state",
-        [
-            np.array([0, 0]),
-            np.array([1, 0]),
-            np.array([0, 1]),
-            np.array([1, 1]),
-        ],
-    )
-    @pytest.mark.parametrize("device_wires", [3, 4, 5])
-    @pytest.mark.parametrize("op_wires", [[0, 1], [1, 0], [2, 0]])
-    def test_basis_state_on_wires_subset(self, state, device_wires, op_wires, tol):
-        """Test basis state initialization on a subset of device wires"""
-        dev = SimulatorDevice(device_wires)
-
-        with mimic_execution_for_apply(dev):
-            dev.apply([qml.BasisState(state, wires=op_wires)])
-
-        res = np.abs(dev.state) ** 2
-        # compute expected probabilities
-        expected = np.zeros([2 ** len(op_wires)])
-        expected[np.ravel_multi_index(state, [2] * len(op_wires))] = 1
-
-        expected = dev._expand_state(expected, op_wires)
-
-        assert np.allclose(res, expected, **tol)
-
-    def test_identity_basis_state(self, shots, tol):
```

Comment:
Same

Reply:
This one is actually identical to `test_basis_state` above, but with a different state. So I parametrized that test to squish them into one and avoid copy-pasted tests 😋